tracer/src/Datadog.Trace.Coverage.collector/FiltersHelper.cs

```diff
@@ -0,0 +1,158 @@
+            filters,
+            list =>
+            {
+                var instance = new Matcher();
+                var lstRegex = new List<Regex>();
+                foreach (var filter in list)
+                {
+                    if (filter is null)
+                    {
+                        continue;
+                    }
+
+                    if (!filter.Contains("**"))
+                    {
+                        try
+                        {
+                            lstRegex.Add(new Regex(filter, RegexOptions.Compiled));
+                        }
+                        catch
+                        {
+                            // .
+                        }
+                    }
+
+                    instance.AddInclude(Path.IsPathRooted(filter) ? filter.Substring(Path.GetPathRoot(filter).Length) : filter);
+                }
+
+                return Tuple.Create(instance, (IReadOnlyList<Regex>)lstRegex);
+            });
+
+        var matcher = value.Item1;
+        // https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.filesystemglobbing.matcher?view=dotnet-plat-ext-6.0
+        var globbingResult = matcher.Match(Path.IsPathRooted(sourcePath) ? sourcePath.Substring(Path.GetPathRoot(sourcePath).Length) : sourcePath).HasMatches;
+        if (globbingResult)
+        {
+            return true;
+        }
+
+        foreach (var regex in value.Item2)
+        {
+            if (regex.IsMatch(sourcePath))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public static bool FilteredByAssemblyAndType(string module, string? type, IReadOnlyList<string> filters)
+    {
+        if (filters.Count == 0)
+        {
+            return false;
+        }
+
+        module = Path.GetFileNameWithoutExtension(module);
+        if (module == null)
+        {
+            return false;
+        }
+
+        foreach (var filter in filters)
+        {
+            var typePattern = filter.Substring(filter.IndexOf(']') + 1);
+            var modulePattern = filter.Substring(1, filter.IndexOf(']') - 1);
+            var moduleRegex = new Regex(WildcardToRegex(modulePattern));
+
+            if (typePattern == "*")
+            {
+                if (moduleRegex.IsMatch(module))
+                {
+                    return true;
+                }
+            }
+            else if (type is not null)
+            {
+                var typeRegex = new Regex(WildcardToRegex(typePattern));
+                if (moduleRegex.IsMatch(module) &&
+                    typeRegex.IsMatch(type))
+                {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+
+        static string WildcardToRegex(string pattern)
+        {
+            return "^" + Regex.Escape(pattern).
+                               Replace("\\*", ".*").
+                               Replace("\\?", "?") + "$";
```

Comment:
You're removing the `\`? 🤔 I don't follow why? Also, shouldn't it be `".?"`?

Reply:
Added some filter test from the coverlet repo, since this piece of code is inspired by their code as we want to follow their filter format in this one.